### ISSUE TITLE
Update to latest version of React: injectEmptyComponent undefined

### ISSUE THREAD

**Issue by rogchap** [Yomguithereal/react-blessed#50]
I know the docs say to use react 0.14.0 but it also says ">="
Is there plans to upgrade to a new version of react?
The part that seems broken for me is that:
`ReactInjection.EmptyComponent` no longer has function `injectEmptyComponent` as it now takes a factory function:

``` javascript
var ReactEmptyComponentInjection = {
 injectEmptyComponentFactory: function(factory) {
   emptyComponentFactory = factory;
  },
};

**Comment by Yomguithereal** [comment]
Yes, it would be nice to support v15. Do you think this is the only thing preventing the renderer from working with v15? The only thing a bit tedious would also be to detect the version of React so that the code may work with 14 or 15 independently.

**Comment by Kasu** [comment]
@Yomguithereal Apparently v15 has changed internals quite a lot. _rootNodeID is a globally increased integer now for instance, which pretty much breaks how you do id operations.

**Comment by DaleJefferson** [comment]
@Yomguithereal This project looks really good, I was just about to start building a dashboard using it when I came across this issue. The last comment was in May and I'm using react 0.15 on all my projects now, do we have a plan for supporting react 0.15. I'm happy to help out, do we have a list of issues that need to be resolved?

**Comment by Yomguithereal** [comment]
For that, one would need to go and perform some archaeology within React's code to see what changed concerning the injection of the renderers.
